### PR TITLE
[Lua] indent fixes

### PR DESCRIPTION
**Pull request by Resike**
Indenting fix for nested tables, and other additions and fixes.

Current:

Fixed:

### CODE REVIEW

File: Lua/Indent.tmPreferences [hunk 1]
```diff
@@ -4,13 +4,13 @@
 	<key>name</key>
 	<string>Indent</string>
 	<key>scope</key>
-	<string>source.lua</string>
+	<string>source.lua - string - comment</string>
 	<key>settings</key>
 	<dict>
 		<key>decreaseIndentPattern</key>
-		<string>^\s*(elseif|else|end|\})\s*$</string>
+		<string>(^(?!=.*(then|do).*).*\b(elseif|else|end|until)\b(?!.*(\"|\'|\[\=*\[)).*$|^((?!\{).)*\}\;?.*$)</string>
```
Comment (keith-hall) [inline Lua/Indent.tmPreferences]: it'd be great if you could format these regex patterns over multiple lines (i.e. `(?x)` mode) to make them readable and add some comments explaining what they do

**Comment by keith-hall** [comment]
note that some work on Lua indentation was also proposed in https://github.com/sublimehq/Packages/pull/1289

**Comment by Resike** [comment]
Thanks, gonna merge that too, but I'll actually rethink this PR tomorrow.

**Comment by Resike** [comment]
I reviewed my initial copy-pasted changes from the lua extended package, and also merged the other requested changes from the other PRs:

Adding `until` and `elsei` for decreaseIndentPattern.
`elsei` prevents the indentation jumping back and forth when typing `elseif`.

`do|then|else|elseif|if` should increase pattern despite whatever before them and not just \s*

And fixes for nested tables, and added the fixes from the other lua PRs.

File: Lua/Indent.tmPreferences [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-		<key>increaseIndentPattern</key>
-		<string>^\s*(else|elseif|for|(local\s+)?function|if|repeat|until|while)\b((?!end).)*$|\{\s*$</string>
+		<key>name</key>
+		<string>Indent</string>
+		<key>scope</key>
+		<string>source.lua</string>
+		<key>settings</key>
+		<dict>
+			<key>decreaseIndentPattern</key>
+			<string>(^\s*\b(elsei|elseif|else|end|until)\b.*$|^((?!\{).)*\}\;?.*$)</string>
```
Comment (michaelblyons) [inline Lua/Indent.tmPreferences]: Googling for "Lua elsei" doesn't return anything that looks useful. Is `elsei` supposed to be in there?

Comment (Resike) [reply Lua/Indent.tmPreferences]: `elsei` prevents the indentation jumping back and forth when typing `elseif`.
I doubt anyone would use elsei as a variable anyway.

**Comment by wbond** [comment]
Thanks for these fixes @Resike, sorry for the long delay in getting them in.

**Pull request merged**